v3/src/components/data-display/models/data-display-layer-model.ts

```diff
@@ -1,15 +1,18 @@
 import {Instance, types} from "mobx-state-tree"
-import { typedId } from "../../../utilities/js-utils"
+import {typedId} from "../../../utilities/js-utils"
 import {DataConfigurationModelUnion} from "./data-configuration-union"
 
 export const kUnknownLayerModelType = "unknownLayer"
 
 export const DataDisplayLayerModel = types.model("DataDisplayLayerModel", {
-    id: types.optional(types.identifier, () => typedId("LAYR")),
-    type: types.optional(types.string, kUnknownLayerModelType),
-    dataConfiguration: types.optional(DataConfigurationModelUnion,
-      () => { throw "dataConfiguration must be overridden" }),
-  })
+  id: types.optional(types.identifier, () => typedId("LAYR")),
+  layerIndex: types.optional(types.number, 0),
```

Comment:
Do we need to store the `layerIndex` in the document, e.g. could it be `volatile` state?